PersonStageThree/src/com/beyond/person/core/model/Dancer.java

```diff
@@ -0,0 +1,48 @@
+package com.beyond.person.core.model;
+
+import com.beyond.person.core.exception.InvalidTypeException;
+import com.beyond.person.core.utils.DanceType;
+
+public class Dancer extends BasePerson {
+    private String groupName;
+    protected String danceType;
+
+    /**
+     * constructor for dancer
+     * @param name
+     * @param lastName
+     * @param groupName
+     * @param designation
+     */
+    public Dancer(String name, String lastName, String groupName, String designation) {
+        super(name, lastName);
+        this.groupName = groupName;
+        this.designation = designation;
+    }
+
+    @Override
+    public void setDesignation(String designation) {
+        try{
+            if(designation.equalsIgnoreCase(String.valueOf(DanceType.LATINO)) || designation.equalsIgnoreCase(String.valueOf(DanceType.ARMENIAN))
```

Comment:
@maria-salmasian throw exception in setter not good idea, move it on service layer

Reply:
changed it